crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -129,26 +153,125 @@ impl C8yMapperActor {
         mqtt_publisher: LoggingSender<MqttMessage>,
         timer_sender: LoggingSender<SyncStart>,
         bridge_status_messages: SimpleMessageBox<MqttMessage, MqttMessage>,
+        message_handlers: HashMap<ChannelFilter, Vec<LoggingSender<MqttMessage>>>,
     ) -> Self {
         Self {
             converter,
             messages,
             mqtt_publisher,
             timer_sender,
             bridge_status_messages,
+            message_handlers,
         }
     }
 
     async fn process_mqtt_message(&mut self, message: MqttMessage) -> Result<(), RuntimeError> {
-        let converted_messages = self.converter.convert(&message).await;
+        // If incoming message follows MQTT topic scheme v1
+        if let Ok((_, channel)) = self.converter.mqtt_schema.entity_channel_of(&message.topic) {
+            match self.converter.try_register_source_entities(&message).await {
+                Ok(pending_entities) => {
+                    self.process_registered_entities(pending_entities, &channel)
+                        .await?;
+                }
+                Err(err) => {
+                    self.mqtt_publisher
+                        .send(self.converter.new_error_message(err))
+                        .await?;
+                    return Ok(());
+                }
+            }
+
+            if !channel.is_entity_metadata() {
+                self.process_data_message(message.clone()).await?;
+            }
+        } else {
+            self.convert_and_publish(&message).await?;
+        }
+
+        Ok(())
+    }
+
+    async fn process_registered_entities(
```

Comment:
It's not easy to follow the progression of a message moving all these methods:

- `process_mqtt_message`
- `process_registered_entities`
- `process_registration_message`
- `process_data_message`
- `convert_and_publish`
- `publish_message_to_subscribed_handles`
- `publish_message`

Doc comments would help to grasp what is the happy path for a message.